src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -29,6 +29,7 @@
 from knack.log import get_logger
 from knack.prompting import prompt_y_n
 from knack.prompting import NoTTYException
+from azure.cli.command_modules.role import graph_client_factory
```

Comment:
With this change to use MS Graph, does the customer/client need different permissions?  Specifically, is it possible for this to fail, for someone for whom it used to work before?  This will become a regression - hence asking.

Reply:
The customer will need MS Graph permission for the extension to get the object ID for them if they don't want to pass it in. There is no way around this in an automated way without using MS Graph, since AAD Graph is deprecated. The behavior will be the same, in that if they don't have permission, it will try to use the one passed in, and if the customer didn't pass anything in, then Custom Locations RP won't be enabled. The AAD Graph delegated user permissions should be automatically converted to the equivalent MS Graph user permissions. This call specifically uses the list API, and the permissions are specified below:

https://learn.microsoft.com/en-us/graph/api/serviceprincipal-list?view=graph-rest-1.0&tabs=http#permissions